legacy.go

```diff
@@ -307,6 +307,16 @@ func (r *legacyLayerReader) Read(b []byte) (int, error) {
 	return r.backupReader.Read(b)
 }
 
+func (r *legacyLayerReader) Seek(offset int64, whence int) (int64, error) {
+	if r.backupReader == nil {
+		if r.currentFile == nil {
+			return 0, errors.New("no current file")
+		}
+		return r.currentFile.Seek(offset, whence)
+	}
+	return 0, errors.New("seek not supported on this stream")
```

Comment:
I could expose it publicly, but generic code needs to do a dummy seek either way since e.g. `*os.File` implements `Seek` but the underlying OS file might be a pipe. So I don't see a great reason to do this.

Regarding adding it to `IsNotSupported`, I don't think we want to mix the storage code error handling with the compute code error handling.